### ISSUE TITLE
optimizer.optimize works - Mariginals does not. Matlab

### ISSUE THREAD

**Issue by etamarlu** [borglab/gtsam#1611]
## Description
The attached Matlab code is a snapshot from trying to convert matlab's [Factor Graph example](https://www.mathworks.com/help/nav/ug/monocular-visual-inertial-odometry-using-factor-graph.html), to work with GTSAM functions.
[VioGTSAM_V2.zip](https://github.com/borglab/gtsam/files/12456767/VioGTSAM_V2.zip)

The error arises after the first map initialization is done using epipolar geometry. 
All factors and initial estimations are created, optimizations is done using either LM solver or isam.calculateEstimate. 

Marginals fail to run with 'Indeterminant linear system' error

## Steps to reproduce
add debug point at line 391, and then execute the line to reproduce the error.

## Additional information
I saw similar cases in [tracking issue #121](https://github.com/borglab/gtsam/issues/121), I couldn't understand the solution

**Comment by etamarlu** [comment]
following #1179 I managed to get it to work, the results however don't look right (uncertainties are too big).

I adjusted the code to work with ISAM, it runs 100 iterations with ISAM, but at some point isam.update returns the same 'Indeterminant' error. The error is about a point that was seen many iterations before, but for only 2 frames.

I leave this post open for now, I dig into the papers, maybe it will help me to better rephrase my initial question. 

Would appreciate any suggestions on relevant reading material
